app/src/main/java/io/homeassistant/companion/android/controls/LightControl.kt

```diff
@@ -38,6 +39,17 @@ class LightControl {
                     PendingIntent.FLAG_CANCEL_CURRENT
                 )
             )
+
+            // On HA Core 2021.5 and later brightness detection has changed
+            // to simplify things in the app lets use both methods for now
+            var supportsBrightness = false
+            val supportedColorModes = entity.attributes["supported_color_modes"] as? List<String>
+            if (supportedColorModes != null) {
+                for (item in supportedColorModes) {
+                    if (item !in NO_BRIGHTNESS_SUPPORT)
+                        supportsBrightness = true
+                }
+            }
```

Comment:
```suggestion
            val supportedColorModes = entity.attributes["supported_color_modes"] as? List<String>
            val supportsBrightness = if (supportedColorModes == null) false else !(supportedColorModes - NO_BRIGHTNESS_SUPPORT).isEmpty()
```

Reply:
I think the syntax is correct?

Reply:
yup just tested it locally and it still works :) even when the condition below is modified thank you!